### ISSUE TITLE
full song with dnn pitch

### ISSUE THREAD

**Issue by MaxMax2016** [PlayVoice/VI-SVS#8]
pitch is no easy to predict

https://user-images.githubusercontent.com/16432329/174222259-257a959d-5e77-4f43-baea-3026482982ee.mp4

**Comment by dtx525942103** [comment]
推理代码中存在的重大BUG
slurs = torch.LongTensor(labels_uvs)
应该修复为
slurs = torch.LongTensor(labels_slr)

**Comment by dutchsing009** [comment]
Oh wow ! , that's indeed way more better and smoother than the other one , so what is missing now for the implementation to be fully done ? can you put the missing stuff in like a code summary , as sometimes I think we are lost in the translation from Chinese to English :) , so i see if i can help you or get help. I saw this today, not sure if that would work for SVS tho but here it is https://github.com/yl4579/PitchExtractor also that https://github.com/r9y9/pyreaper

**Comment by dtx525942103** [comment]
yes you can use https://github.com/yl4579/PitchExtractor but its input is textencoder output
![raw_visinger](https://user-images.githubusercontent.com/16432329/174245332-1ab1567f-82f0-4654-9928-eec43d69f712.png)
![vits_svs](https://user-images.githubusercontent.com/16432329/174245353-ddd73e8b-34f5-44d7-bfa9-d33e60cc36c4.png)

**Comment by Lukelluke** [comment]
> pitch is no easy to predict
> 
>  nn_pit_song.mp4

您好，请问 这个nn版本 pitch 的infer，是采用哪个infer脚本，是“pyton vsinging_song.py”吗？
如果不是，想要达到您提供的这个效果，需要怎么做？望不吝赐教！

**Comment by dtx525942103** [comment]
@Lukelluke 简单的讲，就是使用帧级别的phone,midi通过transformer预测帧级别的pitch，相关pitch预测可以参考
https://github.com/MoonInTheRiver/DiffSinger/blob/master/modules/fastspeech/pe.py
https://github.com/yl4579/PitchExtractor

**Comment by dutchsing009** [comment]
@dtx525942103 Hi ! I had 2 questions if you may. Would it work if the dataset had more than 1 singer , even if it is 2 singers only ? and if it will work. what should be changed besides the "n_speakers" and adding speaker numbers at the training and validation.txt ?? the second one is are you gonna add any updates or you just closed the project and finished it like that?
